File: solutions/cpp/diamond/1/diamond.cpp
#include "diamond.h"

#include <string>
#include <vector>
using namespace std;

namespace diamond {
    vector<string> rows(char c){
        int m = c - 'A';
        int nr = 2*m +1;
        vector<string> diamond;
        string str;
        for(int i=0; i<nr; i++) {
            int idx = i<=m ? i : 2*m - i;
            char cur = static_cast<char>(idx + 'A');
            for(int j= -m; j<= m; j++){
                str += (j==-idx || j==idx) ? cur : ' ';
            }
            diamond.push_back(str);
            str.clear();
        }
        return diamond;
    }
}  // namespace diamond

File: solutions/dart/diamond/1/lib/diamond.dart
class Diamond {
  List<String> rows(String a){
    final aa = 'A'.codeUnitAt(0);
    int idx = a.toUpperCase().codeUnitAt(0)- aa;

    List<String> diamond = [];
    for(var i=0; i<=idx; i++){
      String current = '';
      
      for(var k=0; k<2*idx + 1; k++){    
        if(i==0 && k==idx){
          current += 'A';
        }
        else if(k==idx-i || k==idx+i)
          current += String.fromCharCode(i + aa);
        else
          current += ' ';
      }
      diamond.add(current);         
    }
    for(var i=idx-1; i>=0; i--){
      diamond.add(diamond[i]);
    }
    return diamond;
  }
 }
